### PR TITLE
add "and" and "or" aggregators

### PR DESCRIPTION
**Pull request by mccanne**

### CODE REVIEW

**Review by henridf — APPROVED** [review]
I would suggest adding a test for `and`/`or` to https://github.com/brimsec/zq/blob/master/reducer/reducer_test.go#L73 which exercises the `xPart()` calls in multiple configurations (and likewise for collect/union but I missed those trains...)

**Comment by mccanne** [comment]
> I would suggest adding a test for `and`/`or` to https://github.com/brimsec/zq/blob/master/reducer/reducer_test.go#L73 which exercises the `xPart()` calls in multiple configurations (and likewise for collect/union but I missed those trains...)

Will do but I prefer it as a ztest... coming up!

**Comment by mccanne** [comment]
Just realize I had the spill trigger wrong in the union PR.  I'll fix and make sure both spill tests work exercise the partials path of union/logical/collect.

**Comment by henridf** [comment]
> Will do but I prefer it as a ztest... coming up!

if a ztest can cover the conditions that reducer_test.go, I will be pleased to eat my hat.

**Comment by mccanne** [comment]
> > Will do but I prefer it as a ztest... coming up!
> 
> if a ztest can cover the conditions that reducer_test.go exercises, I will be pleased to eat my hat.

My bad for not writing tests that covered the partials path on union and collect.  See PR #1503.  Do you take sauce on your hat?

**Comment by henridf** [comment]
@mccanne: no hats have been served yet... this doesn't exercise all the conditions covered in reducer_test.go.

File: reducer/logical.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	} else {
+		l.val = l.val || val
+	}
+}
+
+func (l *Logical) Result() zng.Value {
+	if l.val {
+		return zng.True
+	}
+	return zng.False
```
Comment (henridf) [inline reducer/logical.go:43]:
```suggestion
	return l.val
```

**Review by henridf — APPROVED** [review]
The test locks in `-limit` which is currently undocumented and I believe unused. Pointing it out to be sure but if that's ok and intentional, 👍

**Comment by henridf** [comment]
> @henridf: RE: "undocumented"... is this is a reference to user-facing ZQL docs?

@philrz yes that's what I meant by "undocumented". I wouldn't suggest putting it in the docs at this point since I don't see a use for it. I think it's more of an "advanced tuning" option if anything.

**Comment by philrz** [comment]
Thanks @henridf. That's what I guessed, but wanted to be sure. I know we discussed `-sortmem` recently and reached a similar conclusion. I figure once we have some experience building our own setups that tune all these settings to work at scale we could work on an advanced config guide with best practices for users that actually want to try to run this stuff themselves at scale with proper tunings.

File: reducer/ztests/logical-spill.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+script: |
+  zq -t "t1=or(x>2),t3=and(x>0) by key with -limit 1 | sort ." in.tzng
```
Comment (nwt) [inline reducer/ztests/logical-spill.yaml:2]: Maybe make this a zql test instead of a script test?

**Review by nwt — APPROVED** [review]

File: reducer/logical.go (hunk 1 above)
Comment (mccanne) [reply reducer/logical.go:43]: l.val is a go bool and zng.True and zng.False are zng.Values

**Pull request merged**